pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1800,6 +1801,23 @@ func doActivateTail(ctx *domainContext, status *types.DomainStatus,
 
 	status.State = types.RUNNING
 	domainID, state, err := hyper.Task(status).Info(status.DomainName)
+	// If status is unknown, get info again for some retries and if it will not be activated return error.
+	for retry := 0; retry <= unknownStateRetries; retry++ {
+		ctx.ps.StillRunning(agentName, warningTime, errorTime)
+		if err == nil && state == types.UNKNOWN {
+			log.Warnf("doActivateTail(%v) for %s: state is UNKNOWN, retry %d", status.UUIDandVersion, status.DisplayName, retry)
+			time.Sleep(2 * time.Second)
+			domainID, state, err = hyper.Task(status).Info(status.DomainName)
+		} else {
+			break
+		}
+	}
+
+	// if the state is still unknown after the retries we set an error, because we
+	// cannot guarantee that the domain is running.
+	if state == types.UNKNOWN && err == nil {
+		err = fmt.Errorf("The domain state is still unknown after %d retries", unknownStateRetries)
+	}
```

Comment:
I think this can be simplified to something like:
```suggestion
        var state types.SwState
        // If status is unknown, get info again for some retries and if it will not be activated return error.
        for retry := 0; retry < unknownStateRetries; retry++ {
                ctx.ps.StillRunning(agentName, warningTime, errorTime)

                domainID, state, err = hyper.Task(status).Info(status.DomainName)
                if err != nil || state != types.UNKNOWN {
                        break
                }
                log.Warnf("doActivateTail(%v) for %s: state is UNKNOWN, retry %d", status.UUIDandVersion, status.DisplayName, retry)
                // if the state is still unknown after the retries we set an error, because we
                // cannot guarantee that the domain is running.
                err = fmt.Errorf("The domain state is still unknown after %d retries", unknownStateRetries)
                time.Sleep(2 * time.Second)
        }
```

Reply:
@christoph-zededa 
Thanks for your suggestion. However, the error (err = fmt.Errorf("The domain state is still unknown after %d retries", unknownStateRetries)) should be assigned outside the loop because if the UNKNOWN state changes during the loop and we break out, the error should remain nil to reflect the successful state transition.